lib/Dialect/FIRRTL/FIRRTLFolds.cpp

```diff
@@ -103,6 +104,12 @@ static Optional<APSInt> getExtendedConstant(Value operand, Attribute constant,
   if (destWidth < 0)
     return {};
 
+  // InvalidValue inputs simply read as zero.
+  if (auto result = constant.dyn_cast_or_null<InvalidValueAttr>()) {
+    llvm::errs() << "Treating " << operand << " as zero value\n";
+    return APSInt(destWidth, result.getType().isa<UIntType>());
```

Comment:
Do invalid's work with reset/clock/arrays and other types?  I am pretty sure it works with aggregates, so you might need to check this (and just `return {}`)